kafka replay speed: compressed/uncompressed bytes for MaxBytes in fetch requests (#9204)
grafana/mimir: docs/sources/mimir/configure/configure-kafka-backend.md

---
aliases:
  - ../operators-guide/configure/configure-kafka-backend/
description: Learn how to configure Grafana Mimir to use Kafka for ingest storage.
menuTitle: Kafka
title: Configure Grafana Mimir kafka backend
weight: 66
---

> # Configure Grafana Mimir kafka backend

Reviewer: ```suggestion
# Configure the Grafana Mimir Kafka backend
```

> # Configure the Grafana Mimir Kafka backend

Author: @tacole02 I think this doesn't recognize Kafka as a product name. Do you know where we need to make a change for the linter to expect "Kafka" and not "kafka"?

> # Configure the Grafana Mimir Kafka backend

Reviewer: That's a good question, @dimitarvdimitrov , I'll defer to @jdbaldry.

> # Configure the Grafana Mimir Kafka backend

Author: i'll merge this now because it's still not going to the `main` branch and I need the change. But we should figure this out before we merge to `main` (or go with "kafka" instead of "Kafka")

Grafana Mimir supports using Kafka for the first layer of ingestion. This is an experimental feature released in Mimir 2.14.
This page is incomplete. It will be updated as the ingest storage feature matures and moves out of the experimental phase.

## Different Kafka backend implementations

Some Kafka-compatible implementations have different behavior for the Kafka API.

> In order to set up Mimir to work with different Kafka backends you need to configure some parameters.

Reviewer: ```suggestion
To set up Mimir to work with different Kafka backends, you need to configure some parameters.
```

> Below are the kafka flavours and extra configurations needed to set up Mimir with them.

Reviewer: ```suggestion
Here are the Kafka flavors and additional configurations needed to set them up in Mimir.
```

### Apache Kafka

> The default options work with Apache Kafka. No extra configuration is needed.

Reviewer: ```suggestion
Use the default options with Apache Kafka. No additional configuration is needed.
```

### Confluent Kafka

> The default options work with Confluent Kafka. No extra configuration is needed.

Reviewer: ```suggestion
Use the default options with Confluent Kafka. No additional configuration is needed.
```
